src/compas/datastructures/assembly/assembly.py

```diff
@@ -30,10 +30,9 @@ class Assembly(Datastructure):
 
     def __init__(self, name=None, **kwargs):
         super(Assembly, self).__init__()
-        self.attributes = {'name': name or 'Assembly'}
+        self.attributes = {"name": name or "Assembly", "guid_key_map": {}}
```

Comment:
are you adding this here because of serialization?
if so, it will not help, because, currently, the guids are not part of the serialized data of a data object...

in addition, i find it a bit weird to "hide" such a key component of the mapping infrastructure in the "general attributes" dict

Reply:
you're right about hiding it, I will rework this, but I'm confused about the guids not being part of serialized objects: does that mean as a general rule that guids are only meaningful while in-memory but are not intended to be stable across serialization?

Reply:
yes that is currently unfortunately indeed the case

Reply:
we could add it to the serialized version of the object, like we do with `dtype`, which is also not part of the data dict...

Reply:
it would solve a lot of issues we are having with the serialization of state and pull/push to the cloud (speckle)

Reply:
addressed the issue at https://github.com/compas-dev/compas/pull/1032/commits/b589f0ef4e0f7a978e9c49fba49feeb75a53a31e 
now it does not depend on guid serialization anymore and mapping dict is rebuilt on deserialization but not in the custom attrs